services/skus/handler/handler.go

```diff
@@ -67,3 +77,55 @@ func (h *Order) Create(w http.ResponseWriter, r *http.Request) *handlers.AppErro
 
 	return handlers.RenderContent(ctx, order, w, http.StatusCreated)
 }
+
+func (h *Order) CreateNew(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+	raw, err := io.ReadAll(io.LimitReader(r.Body, reqBodyLimit10MB))
+	if err != nil {
+		return handlers.WrapError(err, "Failed to read request body", http.StatusBadRequest)
+	}
+
+	req := &model.CreateOrderRequestNew{}
+	if err := json.Unmarshal(raw, req); err != nil {
+		return handlers.WrapError(err, "Failed to deserialise request", http.StatusBadRequest)
+	}
+
+	ctx := r.Context()
+
+	if err := h.valid.StructCtx(ctx, req); err != nil {
+		verrs, ok := collectValidationErrors(err)
+		if !ok {
+			return handlers.WrapError(err, "Faled to validate request", http.StatusBadRequest)
+		}
+
+		return &handlers.AppError{
+			Message: "Validation failed",
+			Code:    http.StatusBadRequest,
+			Data:    map[string]interface{}{"validationErrors": verrs},
+		}
+	}
+
+	lg := logging.Logger(ctx, "payments").With().Str("func", "CreateOrderNew").Logger()
+
+	result, err := h.svc.CreateOrder(ctx, req)
+	if err != nil {
+		lg.Error().Err(err).Msg("failed to create order")
+
+		return handlers.WrapError(err, "Couldn't finish creating order", http.StatusInternalServerError)
```

Comment:
Just wondering if there is a better alternative to returning the underlying err in the wrap error call. Could we not just return a generic error e.g.

return handlers.WrapError(errors.New("internal server error"), "Error creating the order in the database", http.StatusInternalServerError) 

particularly for this one which seems to be a catch all which could include external system calls.

Reply:
Hm, we can do that. How about `errors.New("something went wrong")`? 😸  I like this way of saying it.

Reply:
Added this. Now returning `errSomethingWentWrong` as a catch-all error.

However, I decided to keep the message as `Couldn't finish creating order` instead of the old `Error creating the order in the database`.

Rationale: `CreateOrder` might return an error after the order have been created in the database, and even after it have been created in Stripe. For instance, if something terrible happened while storing metadata in the end of the method. So the old message was not only imprecise, but misleading.

The new text is not ideal, but at least it implies that the order might have been created, which is closer to what's really going on. Even if it fails at Stripe, it still remains in our database, because we don't roll back (but we absolutely should!).